### ISSUE TITLE
[FEATURE REQUEST]Does the GUI need to be reloaded when pressing the "Start Bench" or "Stop Bench"

### ISSUE THREAD

**Issue by KirikauKiwi** [DeeEmm/DIY-Flow-Bench#273]
I would like to know if it is necessary to reload the GUI when either of the "Start Bench" or "Stop Bench" buttons are pressed in the GUI?
It would feel smoother to the user if the page didn't reload. 

**For me when the page reloads it goes from what I am viewing.**

<img width="1439" alt="Screenshot 2024-12-01 at 12 07 09" src="https://github.com/user-attachments/assets/a1e0ccc9-a012-4fc1-aceb-3689b72c76a8">

**To this after the page reload**

<img width="1439" alt="Screenshot 2024-12-01 at 12 08 34" src="https://github.com/user-attachments/assets/75451427-ff87-443c-b928-19ff32ccd820">

**Comment by DeeEmm** [comment]
It should not be necessary. It may be redirecting within the event defined in webserver.cpp

**Comment by KirikauKiwi** [comment]
Is it not the, window.location.href = '/' in index.html code snippet below that is doing the page reload? It is the same for the Stop Bench button.
```
  document.getElementById('on-button').addEventListener('click', function(){
    console.log('Bench On');
    xhr.open('GET', '/api/bench/on');
    xhr.onload = function() {
      if (xhr.status === 200) window.location.href = '/';
    };
    xhr.send();
```

**Comment by DeeEmm** [comment]
Yes, that line is basically saying that if the event receives a http 200 response (load okay), then redirect the page.

This looks like it is in the javascript file,  I was originally referring to the event definitions in webserver.cpp which is the other end of this transmission...

```
  // API request handlers [JSON confirmation response]
  server->on("/api/bench/on", HTTP_GET, [](AsyncWebServerRequest *request){
      Messages _message;
      Hardware _hardware;
      _message.Handler(language.LANG_BENCH_RUNNING);
      _message.debugPrintf("Bench On \n");
      _hardware.benchOn(); 
      request->send(200, "text/html", "{\"bench\":\"on\"}"); 
      });
```

However looking at that now I can see that part does not redirect (some do)

It should be enough to just comment out the onload() function...

```
  document.getElementById('on-button').addEventListener('click', function(){
    console.log('Bench On');
    xhr.open('GET', '/api/bench/on');
    // xhr.onload = function() {
    //   if (xhr.status === 200) window.location.href = '/';
    // };
    xhr.send();
```

**Comment by DeeEmm** [comment]
Fixed in Build 24120302
